x-pack/plugins/ml/public/application/explorer/actions/load_explorer_data.ts

```diff
@@ -133,172 +131,167 @@ const loadExplorerDataProvider = (
       viewByPerPage,
     } = config;
 
-    const combinedJobRecords: Record<string, CombinedJob> = selectedJobs.reduce((acc, job) => {
-      return { ...acc, [job.id]: mlJobService.getJob(job.id) };
-    }, {});
+    return from(anomalyExplorerChartsService.getCombinedJobs(selectedJobs.map((j) => j.id))).pipe(
```

Comment:
I just realized it should not be fetched here because it causes unnecessary API calls. This function is only used on the anomaly explorer page, so let's keep `mlJobService` with cached jobs for now.
Ideally, there should be a service responsible for propagating selected jobs data (via React state of observable), but we can keep this refactoring out of this PR.